hivemq-edge/src/main/java/com/hivemq/api/resources/impl/ProtocolAdaptersResourceImpl.java

```diff
@@ -576,4 +577,61 @@ public Response getTagSchema(String protocolId) {
             }
         }
     }
+
+    @Override
+    public Response addCompleteAdapter(
+            final String adapterType,
+            final String adapterName,
+            final AdapterConfigModel adapter) {
+        final Optional<ProtocolAdapterInformation> protocolAdapterType =
+                protocolAdapterManager.getAdapterTypeById(adapterType);
+        if (protocolAdapterType.isEmpty()) {
+            return ApiErrorUtils.notFound("Adapter Type not found by adapterType");
+        }
+        final ApiErrorMessages errorMessages = ApiErrorUtils.createErrorContainer();
+        final String adapterId = adapter.getAdapter().getId();
+        final Optional<ProtocolAdapterWrapper<? extends com.hivemq.adapter.sdk.api.ProtocolAdapter>> instance =
+                protocolAdapterManager.getAdapterById(adapterId);
+        if (instance.isPresent()) {
+            System.out.println("FU1");
+            ApiErrorUtils.addValidationError(errorMessages, "id", "Adapter ID must be unique in system");
+            return ApiErrorUtils.badRequest(errorMessages);
+        }
+        validateAdapterSchema(errorMessages, adapter.getAdapter());
+        if (ApiErrorUtils.hasRequestErrors(errorMessages)) {
+            System.out.println("FU2");
+            return ApiErrorUtils.badRequest(errorMessages);
+        }
+        try {
+
+            //TODO: we need to move all these conversions into a common place
+            final Map<String, Object> config =
+                    objectMapper.convertValue(adapter.getAdapter(), new TypeReference<>() {
+                    });
+
+            final List<Map<String, Object>> domainTags = adapter.getDomainTagModels()
+                    .stream()
+                    .map(dtm -> DomainTag.fromDomainTagEntity(dtm, adapterId))
+                    .map(DomainTag::toTagMap)
+                    .collect(Collectors.toList());
+
+            System.out.println(config);
+            System.out.println(domainTags);
+            protocolAdapterManager.addAdapter(adapterType, adapterId, config, domainTags);
+        } catch (final IllegalArgumentException e) {
+            if (e.getCause() instanceof UnrecognizedPropertyException) {
+                e.printStackTrace();
+                ApiErrorUtils.addValidationError(errorMessages,
+                        ((UnrecognizedPropertyException) e.getCause()).getPropertyName(),
+                        "Unknown field on adapter configuration");
+            }
+            e.printStackTrace();
```

Comment:
pls look for any souts and e.printStackTraces in the code

Reply:
done